opal/class/opal_include_list.h

```diff
@@ -0,0 +1,50 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2024      Google, LLC. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#if !defined(OPAL_INCLUDE_LIST_H)
+#define OPAL_INCLUDE_LIST_H
+
+#include "opal_config.h"
+#include "opal/class/opal_object.h"
+#include "opal/class/opal_serializable.h"
+
+/**
+ * An include list. These are strings of the form:
+ *   foo,bar,baz    (include)
+ *   ^foo,bar,baz   (exclude)
+ */
+struct opal_include_list {
+    opal_serializable_t super;
```

Comment:
I question the need for the intermediary `opal_serializable_t` class.

Reply:
It is not strictly necessary but allows the building of additional variable types as needed. I can't think of anything off the top of my head that would benefit but figured it is a good to have.

Reply:
The general idea is that these variables are strings to the user (command line, MPI_T, env vars, etc) but may be something else inside Open MPI: argv list, struct, etc. The serializable takes the string and fills in the internal structure and vise versa. All of this is handles in mca/var so it works with MPI_T as well.